Resources/Private/Fusion/Integration/Header.fusion

```diff
@@ -18,7 +18,13 @@ prototype(Neos.Demo:Integration.Header.Renderer) < prototype(Neos.Demo:Presentat
         maximumLevels = 2
     }
 
-    currentLanguage = ${site.context.dimensions.language[0]}
+    currentLanguage = ${Neos.Dimension.currentValue(site, 'language')}
+    languages = Neos.Fusion:Map {
+        items = ${Neos.Dimension.all(site)['language'].values.iterator}
```

Comment:
This works but there are two issues I have with this line of code:

- In neos/neos-development-collection#4085 we removed the `findPotentialDimensionValues` function but this is now the second time of the usage `${Neos.Dimension.all(site)['language']}` so I guess we should reintroduce it. Maybe with a better naming ;)
- If we have a collection object at hand (`values` is of type `ContentDimensionValues`) do we want to see this usage of `iterator` to work with a e.g. `Neos.Fusion:Map` or similar Implementations which as of now only accept an `array`.

Any thoughts @skurfuerst / @bwaidelich / @mficzel ?

Reply:
1. I also think we will introduce a permutation of "findPotentialDimensionValues" (all values) again eventually. Probably in distinction to "findDimensionRootValue" (values without fallback).
2. We should make `Map` accept an `iterator` that does not implement `countable`. The only value we cannot calculate in this case is `iterator.isLast`.

Reply:
I didn't look into it very closely, but IMO this Fusion code is _far_ too low level.
Couldn't we add a dedicated helper, like
```
languages = ${Neos.Dimension.nameToBeFound(site, 'language')}
```
?

Reply:
Looping/Mapping over uncountable collections seems to be rather simple see https://github.com/neos/neos-development-collection/pull/4168 ... however making collections countable would often make sense.

Reply:
(just discussing with @ahaeslich )

suggestion / idea:

```
// "language" would be ContentDimensionId
languages = ${Neos.Dimension.allDimensionValues(site, 'language')}
```

-> should return `ContentDimensionValues` object I'd say -> this is iterable.

Reply:
@skurfuerst yes, I like that a lot!

Reply:
Will add that so we can proceed with this MR and Neos.Seo